addon/services/infinity.js

```diff
@@ -118,13 +145,130 @@ export default Service.extend({
     @method infinityLoad
     @param {Ember.ArrayProxy} infinityModel
     @param {Integer} increment - to increase page by 1 or -1
+    @param {Node} infinityLoaderElem - important when have multiple infinity loader elements on the page
    */
-  infinityLoad(infinityModel, increment) {
-    if (get(infinityModel, '_loadingMore') || !get(infinityModel, '_canLoadMore')) {
+  infinityLoad(infinityModel, increment = 1, infinityLoaderElem) {
+    if (!infinityModel) {
       return;
     }
 
-    return this.loadNextPage(infinityModel, increment);
+    infinityModel = get(this, 'infinityModels').find(model => model === infinityModel);
+    if (infinityModel) {
+      if (get(infinityModel, '_loadingMore') || !get(infinityModel, '_canLoadMore')) {
+        return;
+      }
+
+      // this is duplicated if this method is called from the route.
+      set(infinityModel, '_increment', increment);
+      return this.loadNextPage(infinityModel, increment, infinityLoaderElem);
+    } else {
+      return true;
+    }
+  },
+
+  /**
+    Use the model method in the place of `this.store.query('model')` to
+    initialize the Infinity Model for your route.
+
+    @method model
+    @param {String} modelName The name of the model.
+    @param {Object} options - optional - the perPage and startingPage to load from.
+    @param {Object} boundParamsOrInfinityModel - optional -
+      params on route to be looked up on every route request or
+      instance of InfinityModel
+    @return {Ember.RSVP.Promise}
+  */
+  model(modelName, options, boundParamsOrInfinityModel) {
```

Comment:
This model hooks is similar to what is in the route, but slightly different.

Reply:
What's the key differences? Could we potentially inject and call the service under the hood in the route mixin?

Reply:
Totally.